File: dcp-solutions/2024-12-21/solution.py
```python
def find_subarray(nums, k):
    left = 0
    current_sum = 0
    result = []

    for right in range(len(nums)):
        current_sum += nums[right]

        while current_sum > k and left <= right:
            current_sum -= nums[left]
            left += 1

        if current_sum == k:
            result.append(nums[left:right+1])

    return result
```

File: dcp-solutions/2024-12-21/solution.rb
```ruby
def find_subarray(nums, k)
  left = 0
  sum = 0
  hash = {}

  nums.each_with_index do |num, right|
    sum += num

    while sum - k > 0 && hash.key?(sum - k)
      left += 1
      sum -= nums[left]
      hash.delete(sum)
    end

    if sum == k
      return nums[left..right]
    end

    hash[num] = right
  end

  []
end
```